src/components/MovieList/MovieList.js

```diff
@@ -109,32 +103,35 @@ class MovieList extends HTMLElement {
     $('.item-list').appendChild(noResultMessage);
   }
 
-  makeMovieListTemplate(movieList) {
-    return movieList.reduce((acc, curr) => {
+  makeMovieListTemplate() {
+    return this.#moviesData.movies.reduce((acc, curr) => {
       return (
         acc +
-        `<movie-item id="${curr.id}" title="${curr.title}" imgUrl="${curr.imgUrl}" score="${curr.score}"></movie-item>`
+        `<movie-item id="${curr.id}" title="${curr.title}" imgUrl="${curr.imgUrl}" score="${curr.score}"  genre="${curr.genre}" description="${curr.description}"></movie-item>`
       );
     }, '');
   }
 
-  toggleVisibleButton() {
-    if (this.#movies.movieResultState.isLastPage) {
-      $('#more-button').classList.add('hide-button');
-      return;
-    }
-
-    $('#more-button').classList.remove('hide-button');
-  }
+  observeMovieItem() {
+    const observer = new IntersectionObserver(
+      (entries, observer) => {
+        entries.forEach(entry => {
+          if (entry.isIntersecting) {
+            observer.unobserve(entry.target);
+
+            if (this.#moviesData.isInPageRange) {
+              this.updateMovieList();
+            }
+          }
```

Comment:
```suggestion

        entries.forEach(entry => {
          if (!entry.isIntersecting) {
            return
          }
          
          observer.unobserve(entry.target);

          if (this.#moviesData.isInPageRange) {
            this.updateMovieList();
          }

        }

```

early return으로 isIntersecting라는 개념을 함수 전반에서 기억하지 않고 초반에 처리할 수 있을 것 같아요